ktor-http/common/src/io/ktor/http/URLBuilder.kt

```diff
@@ -160,3 +160,47 @@ data class Url(
 
     companion object
 }
+
+/**
+ * [Url] authority.
+ */
+val Url.authority: String
+    get() = buildString {
+        user?.let {
+            append(it)
+        }
+        password?.let {
+            append(':')
+            append(it)
+        }
+
+        if (isNotEmpty()) {
+            append('@')
+        }
+
+        append(host)
+        append(':')
+        append(port)
+    }
+
+/**
+ * [URLBuilder] authority.
+ */
+val URLBuilder.authority: String
+    get() = buildString {
+        user?.let {
+            append(it)
+        }
+        password?.let {
+            append(':')
+            append(it)
+        }
+
+        if (isNotEmpty()) {
+            append('@')
+        }
+
+        append(host)
+        append(':')
+        append(port)
```

Comment:
Well, we may add `0` here while we don't do the same for `Url` that is always resolved to a specific port

Reply:
@cy6erGn0m, you are right, authority should be composed like that:
```
authority = [userinfo@]host[:port]
```
so if the port if default we should not add it at all.
We could cover it by tests as well.